instance.tf

```diff
@@ -0,0 +1,30 @@
+resource "aws_instance" "web_server" {
+  ami                         = "ami-051dfed8f67f095f5"
+  associate_public_ip_address = true
+  instance_type               = "t3.micro"
+  subnet_id                   = aws_subnet.public[0].id
+  key_name                    = "ec2key"
```

Comment:
Please change the key name to `main` it will then be easier for me to debug your code (if needed later)